readme.md

```diff
@@ -65,10 +69,169 @@ async function response() {
   }
 }
 
-response();
+gateway();
+```
+
+## Payment methods
+
+#### To create a new payment method type of gateway token:
+
+In this case the payment method will be retained with the provided gateway,
+please check the availability of transaction store in the gateways
+
+Gateways list -> http://docs.agilepay.io/#!/gateway
+
+Gateway token -> http://docs.agilepay.io/#!/payment-method-create-gateway-token
+
+`promise` example:
+``` js
+agilePay.paymentMethod().createGatewayToken('gateway-reference', {
+  number: '4111111111111111',
+  expiry_year: 17,
+  expiry_month: 12,
+  cvv: 123,
+  holder_name: 'John Smith',
+})
+  .then(res => res.getBody().token)
+  .catch(res => {
+  // something went wrong
+  });
+```
+
+`async` `await` examples:
+``` js
+(async function () {
+  try {
+    const gatewayToken = await agilePay.paymentMethod().createGatewayToken('gateway-reference', {
+      number: '4111111111111111',
+      expiry_year: 17,
+      expiry_month: 12,
+      cvv: 123,
+      holder_name: 'John Smith',
+    });
+
+// The response will contain a payment method **token** which is used to perform transactions against the payment method
+  const token = gatewayToken.getBody().token;
+
+  } catch (err) {
+    // something went wrong
+  }
+})();
+```
+or
+
+``` js
+async function gatewayToken() {
+  try {
+    const gatewayToken = await agilePay.paymentMethod().createGatewayToken('gateway-reference', {
+      number: '4111111111111111',
+      expiry_year: 17,
+      expiry_month: 12,
+      cvv: 123,
+      holder_name: 'John Smith',
+    });
+
+// The response will contain a payment method **token** which is used to perform transactions against the payment method
+    const token = gatewayToken.getBody().token;
+    return token;
+
+  } catch (err) {
+    // something went wrong
+  }
+}
+
+gatewayToken();
 ```
 
 
+## Transactions
+
+#### Auth (Charge a credit card with a payment method type of gateway token):
+
+`promise` example:
+``` js
+agilePay.transaction()
+  .setGateway('gateway-reference')
+  .setPaymentMethod('payment-method-token')
+  .auth(5000, 'GBP')
+  .then(res => {
+
+    if (res.getBody().successful) {
+      // the authorisationwas successful
+    } else {
+      // the gateway responded with some errors
+      res.getBody().errors;
+    }
+
+  })
+  .catch(res => {
+  // something went wrong
+  });
+
+```
+`async` `await` example:
+``` js
+async function transactionReference() {
+  try {
+    const transaction = await agilePay.transaction()
+    .setGateway('gateway-reference')
```

Comment:
In this scenario the gateway does not have to be set since is using a payment method type of gateway token